src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/StaffRefDataServiceImpl.java

```diff
@@ -525,6 +580,381 @@ public List<UserType> fetchUserTypes() {
     public List<RoleType> getJobTitles() {
         return roleTypeRepository.findAll();
     }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public StaffProfileCreationResponse updateStaffProfile(StaffProfileCreationRequest profileRequest) {
+
+        log.info("{}:: processStaffProfileUpdation starts::",
+                loggingComponentName);
+
+        jsrValidatorStaffProfile.validateStaffProfile(profileRequest,STAFF_PROFILE_UPDATE);
+
+        CaseWorkerProfile caseWorkerProfileToUpdate = checkStaffProfileForUpdate(profileRequest);
+
+
+        CaseWorkerProfile caseWorkerProfile = updateStaffProfiles(profileRequest, caseWorkerProfileToUpdate);
+        StaffProfileCreationResponse response = null;
+
+        if (null != caseWorkerProfile) {
+
+            staffProfileAuditService.saveStaffAudit(AuditStatus.SUCCESS, StringUtils.EMPTY,
+                    caseWorkerProfile.getCaseWorkerId(), profileRequest, STAFF_PROFILE_UPDATE);
+
+            response = StaffProfileCreationResponse.builder()
+                    .caseWorkerId(caseWorkerProfile.getCaseWorkerId())
+                    .build();
+        }
+        return response;
+    }
+
+    private CaseWorkerProfile checkStaffProfileForUpdate(StaffProfileCreationRequest profileRequest) {
+
+        // get all existing profile from db (used IN clause)
+        CaseWorkerProfile caseWorkerProfile = caseWorkerProfileRepo
+                .findByEmailId(profileRequest.getEmailId().toLowerCase());
+        if (caseWorkerProfile == null) {
+            staffProfileAuditService.saveStaffAudit(AuditStatus.FAILURE,PROFILE_NOT_PRESENT_IN_DB,
+                    StringUtils.EMPTY,profileRequest,STAFF_PROFILE_UPDATE);
```

Comment:
Can you please pass the case worker id instead of `StringUtils.EMPTY?`

Reply:
as caseworker is null we don't have id